smart-contract/src/contract.rs

```diff
@@ -223,28 +224,32 @@ fn try_vote(
         return Err(contract_err("member has already voted"));
     }
 
-    // Esure member joined before the proposal was created and determine weight.
-    let weight = if let Some(m) = member {
+    if info.sender == state.admin && proposal.admin_vote != None {
+        return Err(contract_err("admin has already voted"))
+    }
+
+    if let Some(m) = member {
+        // Ensure member joined before the proposal was created and determine weight.
         if m.joined >= proposal.created {
             return Err(ContractError::Unauthorized {});
         }
-        m.weight
-    } else {
-        state.admin_weight
-    };
 
-    // Add weight to 'yes' or 'no' total in join request.
-    match choice {
-        VoteChoice::Yes => {
-            proposal.yes += weight;
-        }
-        VoteChoice::No => {
-            proposal.no += weight;
-        }
-    };
+        // Add weight to 'yes' or 'no' total in join request.
+        match choice {
+            VoteChoice::Yes => {
+                proposal.yes += m.weight;
+            }
+            VoteChoice::No => {
+                proposal.no += m.weight;
+            }
+        };
 
-    // Add message sender to join request 'voted' vector.
-    proposal.voters.push(info.sender);
+        // Add message sender to join request 'voted' vector.
+        proposal.voters.push(info.sender);
+    } else {
+        // Add admin vote separate from member vote result separate from member votes.
```

Comment:
nit this comment is confusing to me

Reply:
haha i forgot to push my earlier change of this, fixed now